PGM/src/main/java/tc/oc/pgm/polls/types/PollNextMap.java

```diff
@@ -1,24 +1,32 @@
-package tc.oc.pgm.polls;
+package tc.oc.pgm.polls.types;
 
-import org.bukkit.Server;
+import com.google.inject.assistedinject.Assisted;
+import com.google.inject.assistedinject.AssistedInject;
 import org.bukkit.command.CommandSender;
 import org.bukkit.entity.Player;
+import tc.oc.api.docs.PlayerId;
 import tc.oc.api.docs.User;
 import tc.oc.commons.bukkit.tokens.TokenUtil;
+import tc.oc.commons.core.chat.Audiences;
 import tc.oc.pgm.map.PGMMap;
 import tc.oc.pgm.match.MatchManager;
+import tc.oc.pgm.polls.Poll;
+import tc.oc.pgm.polls.PollManager;
 
 public class PollNextMap extends Poll {
+
+    public interface Factory {
+        PollNextMap create(CommandSender sender, PGMMap nextMap, PlayerId playerId);
```

Comment:
If the `CommandSender` is the same as the `PlayerId`, maybe just provide the sender and derive the player? or the other way around?

Reply:
I say that because when instantiating this factory, the `senderToPlayer` method is invoked frequently, and might just be more useful to be put inside the construction.